src/viem-config.ts

```diff
@@ -1,11 +1,18 @@
 import { defineChain } from 'viem';
 import { Chain } from './types';
 import { CHAINS } from './generated/chains';
+import { toUpperSnakeCase } from './utils/strings';
+
+export function chainHttpRpcUrlName(chain: Chain) {
+  return `VIEM_HTTP_RPC_URL_${toUpperSnakeCase(chain.alias)}`;
+}
 
 export function chains() {
   return CHAINS.map((chain) => {
     // All chains must have at least a default provider
     const defaultProvider = chain.providers.find((c) => c.alias === 'default')!;
+    const envRpcUrl = process.env[chainHttpRpcUrlName(chain)];
+    const extraRpcUrls = envRpcUrl ? { environment: { http: [envRpcUrl] } } : {};
```

Comment:
Called this "environment" but open to other suggestions. We could extend this further with other keys in the future if we ever want that

Reply:
Makes sense to me